test/mock/adapter/adapter.go

```diff
@@ -1127,15 +1119,192 @@ func (v *adapterApp) openid4vcInitiatePreAuthorizedIssuance(w http.ResponseWrite
 		return
 	}
 
+	var credentialsToSave map[string]json.RawMessage
+	err = json.Unmarshal([]byte(credentials), &credentialsToSave)
+	if err != nil {
+		handleError(w, http.StatusInternalServerError,
+			fmt.Sprintf("failed to parse credentials : %s", err))
+
+		return
+	}
+
+	for ct, credential := range credentialsToSave {
+		err = v.store.Put(getCredStoreKeyPrefix(key, ct), credential)
+		if err != nil {
+			handleError(w, http.StatusInternalServerError,
+				fmt.Sprintf("failed to server configuration : %s", err))
+
+			return
+		}
+	}
+
 	initiateUrl := "openid-initiate-issuance://?issuer=" + url.QueryEscape(issuer) + "&credential_type=" +
-		url.QueryEscape(credType) + "&pre-authorized_code=" + url.QueryEscape(uuid.NewString()) + "&user_pin_required=true"
+		url.QueryEscape(credType) + "&pre-authorized_code=" + url.QueryEscape(preAuthCode) + "&user_pin_required=true"
 
 	err = t.Execute(w, openid4ciDemo{
 		InitiateUrl: initiateUrl,
 		Pin:         pinNumber,
 	})
 }
 
+func (v *adapterApp) openid4vcIssuerTokenEndpoint(w http.ResponseWriter, r *http.Request) {
```

Comment:
so this token endpoint supports only pre-authorized code flow?

Reply:
At this point - yes